tutorials/Intro_to_causal_reasoning_with_Dags_bnlearn.Rmd

```diff
@@ -0,0 +1,220 @@
+---
+title: "Introducing bnlearn package in R"
+author: "Sara Taheri"
+date: "5/2/2019"
+header-includes:
+   - \usepackage{amsmath}
+   - \usepackage{soul}
+output: 
+  html_document:
+    self_contained: true
+    toc: true
+    toc_float: true
+    number_sections: true
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo=TRUE, warning=FALSE, message=FALSE)
+library(bnlearn)
+library(Rgraphviz)
+```
+
+# Overview of the survey data
+
+## Survey data
+Survey data is a data set that contains information about usage of different transportation systems with a focus on cars and trains for different social groups. It includes these factors:
+
+- **Age (A):** It is recorded as *young* (**young**) for individuals below 30 years, *adult* (**adult**) for individuals between 30 and 60 years old, and *old* (**old**) for people older than 60.
+
+- **Sex (S):** The biological sex of individual, recorded as *male* (**M**) or *female* (**F**).
+
+- **Education (E):** The highest level of education or training completed by the individual, recorded either *high school* (**high**) or *university degree* (**uni**).
+
+- **Occupation (O):** It is recorded as an *employee* (**emp**) or a *self employed* (**self**) worker.
+
+- **Residence (R):** The size of the city the individual lives in, recorded as *small* (**small**) or *big* (**big**).
+
+- **Travel (T):** The means of transport favoured by the individual, recorded as *car* (**car**), *train* (**train**) or *other* (**other**)
+
+Travel is the *target* of the survey, the quantity of interest whose behaviour is under investigation.
+
+```{r, fig.width=3, fig.height=3, fig.align='center'}
+# empty graph
+dag <- empty.graph(nodes = c("A","S","E","O","R","T"))
+arc.set <- matrix(c("A", "E",
+                    "S", "E",
+                    "E", "O",
+                    "E", "R",
+                    "O", "T",
+                    "R", "T"),
+                  byrow = TRUE, ncol = 2,
+                  dimnames = list(NULL, c("from", "to")))
+arcs(dag) <- arc.set
+nodes(dag)
+arcs(dag)
+graphviz.plot(dag)
+```
+
+# Conditional independence in Bayesian networks
+
+Using a DAG structure we can investigate whether a variable is conditionally independent from another variable given a set of variables from the DAG. If the variables
+depend directly on each other, there will be a single arc connecting the nodes
+corresponding to those two variables. If the dependence is indirect, there will
+be two or more arcs passing through the nodes that mediate the association.
+
```

Comment:
Create a new section here:

# Conditional Independence and the Global Markov Property

We use the graphical concepts in the previous section to reason about the data generating process we want to model.  Specifically, the graphical concepts represent assumptions of **conditional independence** in the joint distribution over all the variables in the process.

Reply:
After this, go in order of the previous section

1.  V-structures and how you can have parents that are independent, but become dependent when you condition on the child.  Use the "sprinkler example": rain -> sidewalk wetness <- sprinkler.  rain and sprinkler are independent, but they become conditionally dependent when you observe the sidewalk is wet. The probability the sprinkler is on is the unaffected by finding out it is not raining.  However, given the sidewalk is wet, finding out it is not raining tells you the sprinkler must be on.
2. Markov blanket:  Explain that the Markov blanket makes a node conditionally independent of all the other nodes in the system.  That means that if you predict that node, given the nodes in the Markov blanket, adding any other predictors to your model is overfitting, because the Markov blanket contains all the information available about the node.
3.  D-separation:  Introduce the global Markov property.  Use Elements of Causal Inference page 101 as a guide. Here is a start
> We have a data generating process with a joint distribution $P_X$.  We come up with a DAG $G$.  The global Markov property is a property of G.  It means that all of the d-separations in $G$, correspond to conditional independence assumptions in $P_X$.  When we build a causal model that has an underlying causal DAG, we generally assumes the DAG meets the global Markov property.

Then use your $$ { \!\perp\!\!\!\perp}_{G} \textbf{Y } $$ notation like on page 101 in Scholkopf.

Reply:
After this part, add the following about the local Markov property and the Markov factorization property.

> ## Other aspects of the Markov property of DAGs
> 
> In addition to the global Markov property, we have the local Markov property.  A DAG satisfies the local Markov property with respect to a distribution if each node in the graph is conditionally independent of its non-descendants given its parents.  

Demonstrate the local Markov property with a code example.  Use `dsep` and to show a simple case of a node being d-separated from its grandparent given its parent.  Then generate data from the graph and use `ci.test` to show the same result.

> If a DAG has the Markov factorization property with respect to a distribution, then the distribution factorizes according to the ordering of the DAG.

1. Show the factorization of the Survey Model.
2. Show that you can extract this factorization in bnlearn using `modelstring()`

Reply:
I combined this section with the previous section because I think it makes more sense. Instead of generating data, I used the survey data from the data folder.